micronota/db/uniref.py

```diff
@@ -0,0 +1,380 @@
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/README
+'''
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from os.path import join, basename
+from sqlite3 import connect
+from xml.etree import ElementTree as ET
+import gzip
+
+from skbio import read, Sequence
+
+from ..util import _overwrite, _download
+
+
+def prepare_db(out_d, downloaded, force=False,
+               sprot='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.xml.gz',
+               trembl='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.xml.gz',
+               uniref100='ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/uniref100.fasta.gz',
+               id_map='ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/idmapping_selected.tab.gz'):
+    '''Prepare reference database for UniRef.
+
+    Parameters
+    ----------
+    out_d : str
+        The output directory
+    force : boolean
+        Force overwrite the files
+    downloaded : str
+        File directory. If the files are already downloaded,
+        just use the files in the directory and skip _download.
+
+    Notes
+    -----
+    This function creates the following files:
+
+    * ``uniref100_sprot_Archaea.dmnd``
+    * ``uniref100_sprot_Bacteria.dmnd``
+    * ``uniref100_sprot_Viruses.dmnd``
+    * ``uniref100_sprot_other.dmnd``
+    * ``uniref100_trembl_Archaea.dmnd``
+    * ``uniref100_trembl_Bacteria.dmnd``
+    * ``uniref100_trembl_Viruses.dmnd``
+    * ``uniref100_trembl_other.dmnd``
+    * ``uniref100__other.dmnd``
+
+    * ``uniprotkb.db``
+    * ``uniprotkb_id_map.db``: the map of IDs between databases
+    '''
+
+    id_map_fp = join(out_d, 'uniprotkb_id_map.db')
+    id_map_raw = join(downloaded, basename(id_map))
+    metadata_fp = join(out_d, 'uniprotkb.db')
+    sprot_raw = join(downloaded, basename(sprot))
+    trembl_raw = join(downloaded, basename(trembl))
+    uniref100_raw = join(downloaded, basename(uniref100))
+    try:
+        _download(id_map, id_map_raw, overwrite=force)
+        _download(sprot, sprot_raw, overwrite=force)
+        _download(trembl, trembl_raw, overwrite=force)
+        _download(uniref100, uniref100_raw, overwrite=force)
+    except FileExistsError:
+        pass
+
+    create_id_map(id_map_raw, id_map_fp, force)
+
+    prepare_metadata(sprot_raw, metadata_fp, append=force)
+    prepare_metadata(sprot_raw, metadata_fp, append=True)
```

Comment:
ah, that's a typo. fixed.

On Wed, Feb 17, 2016 at 3:13 PM, Jamie Morton <notifications@github.com>
wrote:

> In micronota/db/uniref.py
> <https://github.com/biocore/micronota/pull/43#discussion_r53247869>:
>
> > +    metadata_fp = join(out_d, 'uniprotkb.db')
> > +    sprot_raw = join(downloaded, basename(sprot))
> > +    trembl_raw = join(downloaded, basename(trembl))
> > +    uniref100_raw = join(downloaded, basename(uniref100))
> > +    try:
> > +        _download(id_map, id_map_raw, overwrite=force)
> > +        _download(sprot, sprot_raw, overwrite=force)
> > +        _download(trembl, trembl_raw, overwrite=force)
> > +        _download(uniref100, uniref100_raw, overwrite=force)
> > +    except FileExistsError:
> > +        pass
> > +
> > +    create_id_map(id_map_raw, id_map_fp, force)
> > +
> > +    prepare_metadata(sprot_raw, metadata_fp, append=force)
> > +    prepare_metadata(sprot_raw, metadata_fp, append=True)
>
> why append twice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/43/files#r53247869>.
>